server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -718,6 +718,10 @@ public void onTimeout(TimeValue timeout) {
             writePortsFile("transport", transport.boundAddress());
         }
 
+        if (!ScriptModule.NULL_FOR_MISSING_VALUE)
+            logger.warn("Script: returning 0 for missing numeric values is deprecated. " +
```

Comment:
This should be using a DeprecationLogger, so the message goes to the deprecation log.